lib/datadog/core/utils/network.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require 'ipaddr'
+require_relative '../vendor/ipaddr'
+
+module Datadog
+  module Core
+    module Utils
+      # Common Network utility functions.
+      module Network
+        DEFAULT_IP_HEADERS_NAMES = %w[
+          x-forwarded-for
+          x-real-ip
+          true-client-ip
+          x-client-ip
+          x-forwarded
+          forwarded-for
+          x-cluster-client-ip
+          fastly-client-ip
+          cf-connecting-ip
+          cf-connecting-ipv6
+        ].freeze
+
+        class << self
+          # Returns a raw client IP associated with the request if it was
+          #   retrieved successfully.
```

Comment:
The return value is still `raw` if we consider that we strip the port and zone specifier (In the case of IPV6)

Reply:
The value we return here is always pre-stripped, right? E.g. `raw_ip_from_request_headers` -> `ip_header` -> `parse_ip`, and `parse_ip` always strips.

Whereas in the old code, the "raw ip" had not gone yet through the stripping; so this is why I'm a bit confused with what we mean by "raw" -- I guess my question is, what's the difference between a raw and a non-raw ip on the new code version?

Reply:
Ohh fair point 

I guess it is up to us to decide what is the concept of `raw`:

- It could be without any manipulation (I think this one makes more sense)
- It could mean just the raw value we want to use in the tag value


If we opt for the first, we could rename the API from `raw`, `stripped_ip_from_request_headers` or something like that

Reply:
Yeah intuitively, I would go for the first option -- raw being the extracted value from a header without having gone through any processing yet.